### DISCUSSION TITLE
Cirrus config in Starlark

### DISCUSSION THREAD

**Discussion by abravalheri** [pyscaffold/pyscaffold#503] [Ideas]
The guys from CirrusCI have been doing an amazing job in improving the ways people can re-use CI configurations.
Lately they have implemented a way of writing the Cirrus tasks in a language that is very similar to Python (Starlark): https://cirrus-ci.org/guide/programming-tasks/. More importantly this language allows "importing" stuff (and therefore sharing re-usable functions).

I personally have contributed porting the code for deep cloning in the PyScaffold repository, that is now available in their official [helpers repository](https://github.com/cirrus-modules/helpers/blob/07f12852d34c0fa7b051d2ca0550e9a4b9d1cee7/lib.star#L167).

In the future, it might be a good idea for us to use Starlark too in our templates.

**Comment by FlorianWilhelm** [comment]
I have never heard of it but if it helps to reduce duplicate code and is still easy to read, it's a good thing.
